runelite-client/src/main/java/net/runelite/client/plugins/itemstats/potions/MoonlightPotion.java

```diff
@@ -0,0 +1,138 @@
+import net.runelite.api.Skill;
+import net.runelite.client.plugins.itemstats.Effect;
+import net.runelite.client.plugins.itemstats.SimpleStatBoost;
+import net.runelite.client.plugins.itemstats.StatBoost;
+import net.runelite.client.plugins.itemstats.StatChange;
+import net.runelite.client.plugins.itemstats.StatsChanges;
+import net.runelite.client.plugins.itemstats.delta.DeltaCalculator;
+import net.runelite.client.plugins.itemstats.stats.Stat;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.stream.Stream;
+import static net.runelite.client.plugins.itemstats.Builders.perc;
+import static net.runelite.client.plugins.itemstats.stats.Stats.ATTACK;
+import static net.runelite.client.plugins.itemstats.stats.Stats.DEFENCE;
+import static net.runelite.client.plugins.itemstats.stats.Stats.HERBLORE;
+import static net.runelite.client.plugins.itemstats.stats.Stats.PRAYER;
+import static net.runelite.client.plugins.itemstats.stats.Stats.STRENGTH;
+
+@RequiredArgsConstructor
+public class MoonlightPotion implements Effect
+{
+	private static final Stat[] OFFENSIVE_MELEE_STATS = {
+			ATTACK, STRENGTH
+	};
+
+	final StatBoost prayerRestore = new StatBoost(PRAYER, false)
+	{
+		@Override
+		public int heals(Client client)
+		{
+			int pray = PRAYER.getMaximum(client);
+			int herb = HERBLORE.getValue(client);
+			return herb < 38 ? 0 : (int) Math.max(pray * 0.25, herb * 0.3) + 7;
+		}
+	};
+
+	@Override
+	public StatsChanges calculate(Client client)
+	{
+		StatsChanges changes = new StatsChanges(0);
+		StatChange restoreChange = prayerRestore.effect(client);
+
+		SimpleStatBoost attackBoost = new SimpleStatBoost(ATTACK, true, getMeleeEffect(client, ATTACK));
+		SimpleStatBoost defenceBoost = new SimpleStatBoost(DEFENCE, true, getMeleeEffect(client, DEFENCE));
+		SimpleStatBoost strengthBoost = new SimpleStatBoost(STRENGTH, true, getMeleeEffect(client, STRENGTH));
+
+		changes.setStatChanges(Stream.of(
+			Stream.of(restoreChange),
+			Stream.of(defenceBoost.effect(client)),
+			Stream.of(OFFENSIVE_MELEE_STATS)
+				.map(stat ->
+				{
+					if (stat == ATTACK)
+					{
+						attackBoost.setStat(stat);
+						return attackBoost.effect(client);
+					}
+					if (stat == STRENGTH)
+					{
+						strengthBoost.setStat(stat);
+						return strengthBoost.effect(client);
+					}
+					return null;
+				}))
+			.reduce(Stream::concat)
+			.orElseGet(Stream::empty)
+			.toArray(StatChange[]::new));
+
+		changes.setPositivity(Stream.of(changes.getStatChanges())
+			.map(StatChange::getPositivity)
+			.max(Comparator.naturalOrder()).orElseThrow());
+		return changes;
+	}
+
+	private DeltaCalculator getMeleeEffect(Client client, Stat stat)
+	{
+		return Arrays.stream(MeleeEffect.values())
+			.filter(t -> t.getStat() == stat && client.getBoostedSkillLevel(Skill.HERBLORE) >= t.getLevelRequirement())
+			.max(Comparator.comparingInt(MeleeEffect::getLevelRequirement))
+			.orElseGet(() -> MeleeEffect.NONE).getDeltaCalculator();
```

Comment:
`.orElseGet(() -> MeleeEffect.NONE)` can be `.orElse(MeleeEffect.NONE)` instead.